code/game/objects/items/robot/items/rlf.dm

```diff
@@ -0,0 +1,50 @@
+/obj/item/rlf
+	name = "Rapid Lollipop Fabricator"
+	desc = "A device used to rapidly deploy lollipop."
+	icon = 'icons/mob/robot_items.dmi'
+	icon_state = "rlf"
+
+/obj/item/rlf/afterattack(atom/A, mob/user as mob, proximity)
+	if(!proximity)
+		return
+	if(!isrobot(user))
+		return
+	if(!iscarbon(A))
+		return
```

Comment:
Could maybe bundle this up in a single line check

Reply:
Bad decision, shouldn't do it.

Reply:
Could you elaborate on that?